### DISCUSSION TITLE
raspberry pi pico 32 inputs

### DISCUSSION THREAD

**Discussion by jozio611611** [shanteacontrols/OpenDeck#316] [Q&A]
Hello. is it possible to make a raspberry pi pico have 32 digital or analog inputs? I need that many inputs to convert my organ pedal which has 32 keys. If someone could help me I would be very grateful. I am writing with google translator, sorry for any mistakes.
![image](https://github.com/user-attachments/assets/8edb2245-3300-432e-a60e-82075b30d692)

**Comment by paradajz** [comment]
https://github.com/shanteacontrols/OpenDeck/wiki/Creating-custom-board-variant

**Comment by jozio611611** [comment]


**Reply by jozio611611** [reply]
Hi paradajz It's a bit difficult for me to prepare a new uf2 file, maybe someone will be willing to help me with this, or build a new uf2 file for 32 pin inputs for the organ pedal. Thank you

**Reply by paradajz** [reply]
Stop spamming and asking other people to essentially do your homework. Read the documentation.

**Reply by Jozio611** [reply]
What's wrong with asking for help from someone who already knows how to do it? I have placed the file below, please comment whether I am going in the right direction

**Reply by paradajz** [reply]
Wrong is the fact that you're asking the same question over and over which is called spamming. If you post the same question again I'll just ban you from the project.

**Reply by Jozio611** [reply]
you're right, the question was asked twice in two different posts, but in this post the same question was accidentally copied because I had Google Translator translations in my notebook and I made a mistake, I'm sorry. I will try to further my education

**Comment by jozio611611** [comment]
---
  mcu: "rp2040"
  usb: true
  uart:
    dinMIDI:
      pins:
        rx:
          port: "0"
          index: 1
        tx:
          port: "0"
          index: 0
        tx:
          port: "0"
          index: 4
      components: 64
    buttons:
    type: "native"
    extPullups: false
    pins:
    -
      port: "0"
      index: 2
    -
      port: "0"
      index: 3
    -
      port: "0"
      index: 4
    -
      port: "0"
      index: 5
    -
      port: "0"
      index: 6
    -
      port: "0"
      index: 7
    -
      port: "0"
      index: 8
    -
      port: "0"
      index: 9
    -
      port: "0"
      index: 10
    -
      port: "0"
      index: 11
    -
      port: "0"
      index: 12
    -
      port: "0"
      index: 13
    -
      port: "0"
      index: 14
    -
      port: "0"
      index: 15
    -
      port: "0"
      index: 16
    -
      port: "0"
      index: 17
    -
      port: "0"
      index: 18
    -
      port: "0"
      index: 19
    -
      port: "0"
      index: 20
    -
      port: "0"
      index: 21
    -
      port: "0"
      index: 22
    -
      port: "0"
      index: 23
    -
      port: "0"
      index: 24
    -
      port: "0"
      index: 25
    -
      port: "0"
      index: 26
    -
      port: "0"
      index: 27
    -
      port: "0"
      index: 28
    -
      port: "0"
      index: 29
    -
      port: "0"
      index: 30
    -
      port: "0"
      index: 31
    -
      port: "0"
      index: 32
    test: true
[pico Organ pedal 32 Pins.zip](https://github.com/user-attachments/files/18007509/pico.Organ.pedal.32.Pins.zip)

**Comment by Bosswhite12** [comment]
In the discussion below (see link) This kind guy posted some Multiplex uF2 files they also include info on which pins to use.
One of these worked for me.  Nick

https://github.com/shanteacontrols/OpenDeck/discussions/177#discussioncomment-5063095

**Reply by Jozio611** [reply]
Hi bosswhite12  I've just seen these files in the post you sent me, but I need 32 input risers without a multiplexer, from what I see, it's probably impossible to do it. The pico will not have as many inputs as I need for the organ pedal. I have programmed an Arduino mega 2560 and there are enough pins for my needs, but I wanted to try pico because it is easy to program. Thank you for your interest, and paradajz for an interesting project :)
